### ISSUE TITLE
[SSD/PyTorch] The JSON element does not have the requested type

### ISSUE THREAD

**Issue by ksv87** [NVIDIA/DeepLearningExamples#1422]
Related to **Model/Framework(s)** 
*SSD/PyTorch*

**Describe the bug**
```
DLL 2024-09-05 08:00:14.587129 - PARAMETER dataset path : /coco  epochs : 65  batch size : 64│F1Help  F2Setup F3SearchF4FilterF5List  F6SortByF7Nice -F8Nice +F9Kill  F10Quit
  eval batch size : 32  no cuda : False  seed : None  checkpoint path : None  mode : training├──────────────────────────────────────────────────────────────────────────────────────────────  eval on epochs : [21, 31, 37, 42, 48, 53, 59, 64]  lr decay epochs : [43, 54]  learning rat│ksv@neroksv:~$
e : 0.0026  momentum : 0.9  weight decay : 0.0005  lr warmup : 300  backbone : resnet50  back│
bone path : None  num workers : 8  AMP : True  precision : amp                               │
Using seed = 6772                                                                            │
Loading annotations into memory...                                                           │
Done (t=0.03s)                                                                               │
Creating index...                                                                            │
terminate called after throwing an instance of 'simdjson::simdjson_error'                    │
  what():  The JSON element does not have the requested type.                                │
Aborted (core dumped)  
```

**To Reproduce**
1. Install https://github.com/NVIDIA/DeepLearningExamples/tree/master/PyTorch/Detection/SSD#quick-start-guide
2. Set COCO_DIR to my custom dataset
3. Change 
https://github.com/NVIDIA/DeepLearningExamples/blob/729963dd47e7c8bd462ad10bfac7a7b0b604e6dd/PyTorch/Detection/SSD/ssd/model.py#L61C9-L61C54
to my class count + 1
4. Launch python main.py --backbone resnet50 --warmup 300 --bs 64 --d│    565 root       0.0  0.0  0:00.58 ├─ /lib/systemd/systemd-journald                         ata /coco                              

**Expected behavior**

**Environment**
Please provide at least:
* Container version (22.10-py3):
* GPUs in the system: (1 NVIDIA RTX A5000 - 24GB):
* CUDA driver version (545.23.08):

**Comment by Srikanth-ECE** [comment]
Hi @ksv87,

I investigated the SSD data loading flow:

main.py -> get_train_loader() -> COCOPipeline -> dali.ops.readers.COCO()

From the code, the JSON parsing is performed by NVIDIA DALI's COCO reader, which appears to be where the simdjson exception originates.

Could you please provide:

the full stack trace,
a sample of your custom COCO annotation file (or its structure),
whether the issue also occurs with the official COCO dataset,
and the DALI version you're using?

That would help determine whether this is an SSD issue, a DALI issue, or a dataset-format compatibility issue.

**Comment by 2000pd3rvr** [comment]
Regarding **[SSD/PyTorch] The JSON element does not have the requested type**:
Could you start with a minimal reproduction, including exact package versions and the shortest script that triggers the failure. If a stack trace is available, identify the first frame within the library API rather than application code.

— AMA

**Comment by ksv87** [comment]
I haven't been working on this issue for a long time, and I can't provide any information. If no one else has encountered this error in the past two years, I think we can consider it resolved.
